peer/brontide.go

```diff
@@ -4090,6 +4122,52 @@ func (p *Brontide) handleRemovePendingChannel(req *newChannelMsg) {
 	p.addedChannels.Delete(chanID)
 }
 
+// handlePeerStorageMessage `PeerStorage` messages. It stores the data and waits
+// for `peerSrorageDelay` time before sending back the data was stored as a form
+// of ack.
+func (p *Brontide) handlePeerStorageMessage(msg *lnwire.PeerStorage) {
+	latestBlob, err := p.cfg.PeerStorage.StorePeerBackup(msg.Blob, p.
+		String())
+	if err != nil {
+		p.storeError(err)
+		p.log.Errorf("%v", err)
+		return
+	}
+
+	time.AfterFunc(peerStorageDelay, func() {
+		err := p.SendMessage(true, &lnwire.YourPeerStorage{
+			Blob: latestBlob,
+		})
+		p.storeError(err)
+		p.log.Errorf("%v", err)
+	})
+
+}
+
+func (p *Brontide) handleYourPeerStorageMessage(msg *lnwire.YourPeerStorage) {
+
+	if p.LatestPeerBackup == nil {
```

Comment:
This seems like it ought to be `!=`

Reply:
Correct, note that this untested yet, I just wanted to show a rough design of the implementation.